### PR TITLE
Fix branch distance calculation

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://github.com/dotnet/corefx/issues/26621. We incorrectly calculated branch distance when long branch instruction is used with a short distance.

### CODE REVIEW

**Comment by tmat** [comment]
@nguerrera PTAL

File: src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/Encoding/ControlFlowBuilder.cs [hunk 1]
```diff
@@ -27,25 +27,27 @@ internal BranchInfo(int ilOffset, LabelHandle label, ILOpCode opCode)
...
 
-                distance = labelTargetOffset - (ILOffset + shortBranchSize);
-                if (unchecked((sbyte)distance) == distance)
+                int branchInstructionSize = 1 + (isShortBranch ? sizeof(sbyte) : sizeof(int));
+                int distance = labelTargetOffset - (ILOffset + branchInstructionSize);
+
+                if (isShortBranch && unchecked((sbyte)distance) != distance)
                 {
-                    return true;
+                    // We could potentially implement algortihm that automatically fixes up the branch instructions as well to accomodate bigger distances,
```
Comment (jkotas) [inline src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/Encoding/ControlFlowBuilder.cs]: Typo: algortihm

File: src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/Encoding/ControlFlowBuilder.cs [hunk 2]
```diff
@@ -27,25 +27,27 @@ internal BranchInfo(int ilOffset, LabelHandle label, ILOpCode opCode)
...
+                    // however an optimal algorithm would be rather complex (something like: calculate topological ordering of crossing branch instructions 
+                    // and then use fixed point to eliminate cycles). If the caller doesn't care about optimal IL size they can use long branches whenever the 
+                    // distance is unknown upfront. If they do they probably already implement more sophisticad algorithm for IL layout optimization already.
```
Comment (jkotas) [inline src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/Encoding/ControlFlowBuilder.cs]: Typo: sophisticad

**Comment by tmat** [comment]
test OSX x64 Debug Build  please

**Comment by tmat** [comment]
test OSX x64 Debug Build  please

**Pull request merged**